api/src/main/java/org/pmiops/workbench/db/dao/DataSetServiceImpl.java

```diff
@@ -327,14 +328,21 @@ private String getColumnName(CdrBigQuerySchemaConfig.TableConfig config, String
   @VisibleForTesting
   public ValuesLinkingPair getValueSelectsAndJoins(List<DomainValuePair> valueSetList, Domain d) {
     List<String> values =
-        valueSetList.stream().map(valueSet -> valueSet.getValue()).collect(Collectors.toList());
+        valueSetList.stream()
+            .map(valueSet -> valueSet.getValue().toUpperCase())
+            .collect(Collectors.toList());
     values.add(0, "CORE_TABLE_FOR_DOMAIN");
+
+    if (d == Domain.OBSERVATION) {
```

Comment:
Are all observation domains in survey? If so, is there a reason not to send up Domain.SURVEY from the UI?